### PR TITLE
Change Server to start/stop from create/destroy

### PR DESCRIPTION
**Pull request by addievo**
### Description
The goal of this PR is to replace the server create/destroy with start/stop, as logically, starting and stopping makes more sense in Polykey as other layers such as QUIC and WS operate on start/stop.

### Tasks

- [x] 1. Create new events.
- [x] 2. Replace events.
- [x] 3. Refactor function names

### Final checklist

* [x] Domain specific tests
* [x] Full tests
* [x] Updated inline-comment documentation
* [x] Lint fixed
* [x] Squash and rebased
* [x] Sanity check the final build

### CODE REVIEW

File: src/RPCServer.ts [hunk 1]
```diff
@@ -18,11 +18,11 @@ import type { JSONValue } from './types';
 import type { IdGen } from './types';
 import type { ErrorRPC, ErrorRPCRemote } from './errors';
 import { ReadableStream, TransformStream } from 'stream/web';
-import { CreateDestroy, ready } from '@matrixai/async-init/dist/CreateDestroy';
+import { ready } from '@matrixai/async-init/dist/CreateDestroy';
```
Comment (tegefaulkes) [inline src/RPCServer.ts:21]: Needs to import from `@matrixai/async-init/dist/StartStop`

File: src/RPCServer.ts [hunk 2]
```diff
@@ -44,28 +44,23 @@ const cleanupReason = Symbol('CleanupReason');
...
-   * Creates RPC server.
+   * Starts RPC server.
 
    * @param obj
    * @param obj.manifest - Server manifest used to define the rpc method
    * handlers.
    * @param obj.middlewareFactory - Middleware used to process the rpc messages.
-   * The middlewareFactory needs to be a function that creates a pair of
+   * The middlewareFactory needs to be a function that starts a pair of
    * transform streams that convert `Uint8Array` to `JSONRPCRequest` on the forward
```
Comment (tegefaulkes) [inline src/RPCServer.ts:64]: Did you find and replace this? it's still creating transforms streams, the change doesn't make sense.

File: src/RPCServer.ts [hunk 3]
```diff
@@ -77,7 +72,7 @@ class RPCServer extends EventTarget {
    * the handler to handle timeout before it is forced to end. Defaults to 2,000 milliseconds.
    * @param obj.logger
    */
-  public static async createRPCServer({
+  public static async startRPCServer({
     manifest,
     middlewareFactory = rpcUtilsMiddleware.defaultServerMiddlewareWrapper(),
     handlerTimeoutTime = Infinity, // 1 minute
```
Comment (tegefaulkes) [inline src/RPCServer.ts:78]: The `StartStop` pattern doesn't use a static async factory like this, this needs to be fully removed.

Start stop is used by doing
```ts
const thing = new thing();
await thing.start();
```

Usage of the `RPCServer` needs to be updated in tests to reflect this.

File: src/RPCServer.ts [hunk 4]
```diff
@@ -214,10 +209,10 @@ class RPCServer extends EventTarget {
     this.filterSensitive = filterSensitive || rpcUtils.filterSensitive;
   }
 
-  public async destroy(force: boolean = true): Promise<void> {
+  public async stop(force: boolean = true): Promise<void> {
     // Log and dispatch an event before starting the destruction
-    this.logger.info(`Destroying ${this.constructor.name}`);
-    this.dispatchEvent(new events.EventRPCServerDestroy());
+    this.logger.info(`Stopping ${this.constructor.name}`);
+    this.dispatchEvent(new events.EventRPCServerStopping());
```
Comment (tegefaulkes) [inline src/RPCServer.ts:215]: We shouldn't dispatch a separate event here, it's handled by the `async-init` decorator events now.

File: src/RPCServer.ts [hunk 5]
```diff
@@ -77,7 +72,7 @@ class RPCServer extends EventTarget {
    * the handler to handle timeout before it is forced to end. Defaults to 2,000 milliseconds.
    * @param obj.logger
    */
-  public static async createRPCServer({
+  public static async startRPCServer({
```
Comment (CMCDragonkai) [inline src/RPCServer.ts:75]: We don't use `startRPCServer`. See how `StartStop` is used in `QUICServer`. We use `QUICServer.start`. There is no static function.

Please review `QUICServer` first before asking for another review.

File: src/events.ts [hunk 6]
```diff
@@ -41,6 +41,11 @@ class EventRPCServerCreated extends EventRPCServer {}
 class EventRPCServerDestroy extends EventRPCServer {}
 
 class EventRPCServerDestroyed extends EventRPCServer {}
```
Comment (CMCDragonkai) [inline src/events.ts:43]: Get rid of the old `Destroy` events. They are no longer relevant.

File: src/RPCServer.ts [hunk 7]
```diff
@@ -231,8 +226,8 @@ class RPCServer extends EventTarget {
     }
 
     // Log and dispatch an event after the destruction has been completed
-    this.dispatchEvent(new events.EventRPCServerDestroyed());
-    this.logger.info(`Destroyed ${this.constructor.name}`);
+    this.dispatchEvent(new events.EventRPCServerStopped());
```
Comment (tegefaulkes) [inline src/RPCServer.ts:229]: Same here, we shouldn't dispatch stopped here, that's handled by the `async-init` decorated event now.

File: src/RPCServer.ts [hunk 8]
```diff
@@ -214,10 +209,10 @@ class RPCServer extends EventTarget {
     this.filterSensitive = filterSensitive || rpcUtils.filterSensitive;
   }
 
-  public async destroy(force: boolean = true): Promise<void> {
+  public async stop(force: boolean = true): Promise<void> {
```
Comment (CMCDragonkai) [inline src/RPCServer.ts:212]: Can you change this to `{ force?: boolean }` as per @amydevs asked for already.

File: src/events.ts [hunk 9]
```diff
@@ -41,6 +41,11 @@ class EventRPCServerCreated extends EventRPCServer {}
...
+class EventRPCServerStart extends EventRPCServer {}
+class EventRPCServerStarted extends EventRPCServer {}
+class EventRPCServerStopping extends EventRPCServer {}
+
+class EventRPCServerStopped extends EventRPCServer {}
```
Comment (tegefaulkes) [inline src/events.ts:48]: use proper spacing, there is an empty line between each definition by convention.

**Comment by CMCDragonkai** [comment]
Did you bring in `HandlerTypes`?

**Review by tegefaulkes — CHANGES_REQUESTED** [review]
Some changes are needed.

**Comment by CMCDragonkai** [comment]
Change `RPCClient` to just regular constructor, no need for any `CreateDestroy` anymore. Remove all destroy events.

File: src/RPCServer.ts (hunk 8 above)
Comment (tegefaulkes) [reply src/RPCServer.ts:212]: Its force and reason right? as in `{force = true, reason}: {force?: boolean, reason?: any} = {}`

File: src/RPCServer.ts (hunk 5 above)
Comment (CMCDragonkai) [reply src/RPCServer.ts:75]: @addievo before you resolve a conversation. Put down a comment explaining how you resolved it.

**Pull request merged**